src/Chainweb/Pact/PactService.hs

```diff
@@ -364,41 +364,45 @@ serviceRequests logFn memPoolAccess reqQ = do
             LocalMsg LocalReq{..} -> do
-                tryOne "execLocal" _localResultVar $ execLocal _localRequest
+                trace logFn "Chainweb.Pact.PactService.execLocal" () 0 $
+                    tryOne "execLocal" _localResultVar $
+                        execLocal _localRequest
                 go
             NewBlockMsg NewBlockReq {..} -> do
                 trace logFn "Chainweb.Pact.PactService.execNewBlock"
                     (_parentHeader _newBlockHeader) 1 $
                     tryOne "execNewBlock" _newResultVar $
-                    execNewBlock memPoolAccess _newBlockHeader _newMiner
+                        execNewBlock memPoolAccess _newBlockHeader _newMiner
                 go
             ValidateBlockMsg ValidateBlockReq {..} -> do
                 trace logFn "Chainweb.Pact.PactService.execValidateBlock"
                     _valBlockHeader
                     (length (_payloadDataTransactions _valPayloadData)) $
                     tryOne "execValidateBlock" _valResultVar $
-                    execValidateBlock _valBlockHeader _valPayloadData
+                        execValidateBlock _valBlockHeader _valPayloadData
                 go
             LookupPactTxsMsg (LookupPactTxsReq restorePoint txHashes resultVar) -> do
                 trace logFn "Chainweb.Pact.PactService.execLookupPactTxs" ()
                     (length txHashes) $
                     tryOne "execLookupPactTxs" resultVar $
-                    execLookupPactTxs restorePoint txHashes
+                        execLookupPactTxs restorePoint txHashes
                 go
             PreInsertCheckMsg (PreInsertCheckReq txs resultVar) -> do
                 trace logFn "Chainweb.Pact.PactService.execPreInsertCheckReq" ()
                     (length txs) $
                     tryOne "execPreInsertCheckReq" resultVar $
-                    V.map (() <$) <$> execPreInsertCheckReq txs
+                        V.map (() <$) <$> execPreInsertCheckReq txs
                 go
             BlockTxHistoryMsg (BlockTxHistoryReq bh d resultVar) -> do
-              trace logFn "Chainweb.Pact.PactService.execBlockTxHistory" bh 1 $
-                tryOne "execBlockTxHistory" resultVar $
-                execBlockTxHistory bh d
+                trace logFn "Chainweb.Pact.PactService.execBlockTxHistory" bh 1 $
+                    tryOne "execBlockTxHistory" resultVar $
+                        execBlockTxHistory bh d
+                go
             HistoricalLookupMsg (HistoricalLookupReq bh d k resultVar) -> do
-              trace logFn "Chainweb.Pact.PactService.execHistoricalLookup" bh 1 $
-                tryOne "execHistoricalLookup" resultVar $
-                execHistoricalLookup bh d k
+                trace logFn "Chainweb.Pact.PactService.execHistoricalLookup" bh 1 $
+                    tryOne "execHistoricalLookup" resultVar $
+                        execHistoricalLookup bh d k
+                go
```

Comment:
Adds recursive call to `go`. See above.

Reply:
I know this got merged in the other PR but I'm wondering how we missed this. Wouldn't rosetta tests be failing after the first call?

Reply:
Also, just want to note that this is a weakness in this function, the type of the casing should force code to handle the recursion (or not, ie for `CloseMsg`)